integrationTests/chainSimulator/mempool/testutils_test.go

```diff
@@ -163,12 +162,7 @@ func selectTransactions(t *testing.T, simulator testsChainSimulator.ChainSimulat
 
 	mempool := poolsHolder.ShardDataStore(shardAsString).(*txcache.TxCache)
 
-	selectedTransactions, gas := mempool.SelectTransactions(
-		selectionSession,
-		process.TxCacheSelectionGasRequested,
-		process.TxCacheSelectionMaxNumTxs,
-		process.TxCacheSelectionLoopMaximumDuration,
-	)
+	selectedTransactions, gas := mempool.SelectTransactions(selectionSession)
```

Comment:
Can we keep passing the arguments to the mempool/txpool? 

The logic of how many txs to select could change at runtime depending on how the execution keeps up with the proposals, and I think this logic should be outside of the mempool.

Maybe in a follow up PR?